### DISCUSSION TITLE
Creating multiple connectors out of an element without ports

### DISCUSSION THREAD

**Discussion by seekeramento** [clientIO/joint#1801] [Q&A]
My case is as follows: An element on a diagram can can be connected to many elements. Having specific ports to do the connection among elements makes it unusable because an element can be moved around, and I may end up with connectors that are turning around the element for no reason. I need a connector to start from any place on the element border. At the same time, I am hoping that there is an algorithm that organize the connectors' placement on an element border.

Is there an example that is similar to my scenario?

**Comment by kumilingus** [comment]
If you don't want to use ports with fixed positions, you can define the start/end of the link with [anchor](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#anchors.intro).

Here's an example of `midSide` anchor: https://jsfiddle.net/kumilingus/8tqxksra/

For a custom solution write a [custom anchor](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#anchors.custom).

**Comment by rmoryc** [comment]
This topic seems close to my case but I could not find a solution in JointJS yet. I need also to handle multiple links between the same components. Furthemore:
- anchors are automatically placed on the "best" side of the component (already implemented in JointJS)
- if severals anchors are eventually placed one the same side of the component, they shall be at different places in order to distinguish each link

I have attached an example of what I need to achieve.

![gojs_example](https://github.com/clientIO/joint/assets/16774162/fe6741fa-af48-4de2-b543-a03d1bb4b004)

**Comment by kumilingus** [comment]
I think the best way to approach this is write a function that:
- iterates over each link in the diagram
    - finds the best side for each link end based on the position of both source and target
- iterates over each element and look at all links at given side
    - set the anchor of the link and adapt `dx` and `dy` based on number links
    ```ts
    // e.g.
    link1.prop('source/anchor', { name: 'right', args: { dx: 0, dy: -5 });
    link2.prop('source/anchor', { name: 'right', args: { dx: 0, dy: 5 });
    ```
And run the function every time the user reconnects a link or moves an element.

---

You might also want to check this [example](https://github.com/clientIO/joint/blob/master/packages/joint-core/demo/ports/dynamic.js) with dynamic ports.
